custom/icds/location_rationalization/parser.py

```diff
@@ -0,0 +1,74 @@
+from collections import defaultdict
+
+from custom.icds.location_rationalization.const import (
+    EXTRACT_TRANSITION,
+    MERGE_TRANSITION,
+    MOVE_TRANSITION,
+    SPLIT_TRANSITION,
+    TRANSITION_COLUMN_NAME,
+    VALID_TRANSITIONS,
+)
+
+
+class Parser(object):
+    def __init__(self, worksheet, location_types):
+        self.worksheet = worksheet
+        self.requested_transitions = {}
```

Comment:
After reading more thoroughly, this is a simpler data structure than I thought, maybe doesn't need the paragraphs that are in `_get_condensed_rows`, but it could still use a comment.

Reply:
added some comments in https://github.com/dimagi/commcare-hq/pull/26865/commits/f6e5c860b83dbc7efd89ce6850d1fe9974ef253c